Mathlib/LinearAlgebra/QuadraticForm/Basic.lean

```diff
@@ -353,142 +365,142 @@ def ofPolar (toFun : M → R) (toFun_smul : ∀ (a : R) (x : M), toFun (a • x)
+#align quadratic_form.add_apply QuadraticMap.add_apply
 
-instance : AddCommMonoid (QuadraticForm R M) :=
+instance : AddCommMonoid (QuadraticMap R M N) :=
   DFunLike.coe_injective.addCommMonoid _ coeFn_zero coeFn_add fun _ _ => coeFn_smul _ _
 
-/-- `@CoeFn (QuadraticForm R M)` as an `AddMonoidHom`.
+/-- `@CoeFn (QuadraticMap R M)` as an `AddMonoidHom`.
 
 This API mirrors `AddMonoidHom.coeFn`. -/
 @[simps apply]
-def coeFnAddMonoidHom : QuadraticForm R M →+ M → R where
+def coeFnAddMonoidHom : QuadraticMap R M N →+ M → N where
   toFun := DFunLike.coe
   map_zero' := coeFn_zero
   map_add' := coeFn_add
-#align quadratic_form.coe_fn_add_monoid_hom QuadraticForm.coeFnAddMonoidHom
+#align quadratic_form.coe_fn_add_monoid_hom QuadraticMap.coeFnAddMonoidHom
 
-/-- Evaluation on a particular element of the module `M` is an additive map over quadratic forms. -/
+/-- Evaluation on a particular element of the module `M` is an additive map over quadratic maps. -/
 @[simps! apply]
-def evalAddMonoidHom (m : M) : QuadraticForm R M →+ R :=
+def evalAddMonoidHom (m : M) : QuadraticMap R M N →+ N :=
   (Pi.evalAddMonoidHom _ m).comp coeFnAddMonoidHom
-#align quadratic_form.eval_add_monoid_hom QuadraticForm.evalAddMonoidHom
+#align quadratic_form.eval_add_monoid_hom QuadraticMap.evalAddMonoidHom
 
 section Sum
 
 @[simp]
-theorem coeFn_sum {ι : Type*} (Q : ι → QuadraticForm R M) (s : Finset ι) :
+theorem coeFn_sum {ι : Type*} (Q : ι → QuadraticMap R M N) (s : Finset ι) :
     ⇑(∑ i in s, Q i) = ∑ i in s, ⇑(Q i) :=
   map_sum coeFnAddMonoidHom Q s
-#align quadratic_form.coe_fn_sum QuadraticForm.coeFn_sum
+#align quadratic_form.coe_fn_sum QuadraticMap.coeFn_sum
 
 @[simp]
-theorem sum_apply {ι : Type*} (Q : ι → QuadraticForm R M) (s : Finset ι) (x : M) :
+theorem sum_apply {ι : Type*} (Q : ι → QuadraticMap R M N) (s : Finset ι) (x : M) :
     (∑ i in s, Q i) x = ∑ i in s, Q i x :=
-  map_sum (evalAddMonoidHom x : _ →+ R) Q s
-#align quadratic_form.sum_apply QuadraticForm.sum_apply
+  map_sum (evalAddMonoidHom x : _ →+ N) Q s
+#align quadratic_form.sum_apply QuadraticMap.sum_apply
 
 end Sum
 
-instance [Monoid S] [DistribMulAction S R] [SMulCommClass S R R] :
-    DistribMulAction S (QuadraticForm R M) where
+instance [Monoid S] [DistribMulAction S N] [SMulCommClass S R N] [SMulCommClass R S N] :
+    DistribMulAction S (QuadraticMap R M N) where
   mul_smul a b Q := ext fun x => by simp only [smul_apply, mul_smul]
-  one_smul Q := ext fun x => by simp only [QuadraticForm.smul_apply, one_smul]
+  one_smul Q := ext fun x => by simp only [QuadraticMap.smul_apply, one_smul]
   smul_add a Q Q' := by
     ext
     simp only [add_apply, smul_apply, smul_add]
   smul_zero a := by
     ext
     simp only [zero_apply, smul_apply, smul_zero]
 
-instance [Semiring S] [Module S R] [SMulCommClass S R R] :
-    Module S (QuadraticForm R M) where
+instance [Semiring S] [Module S N] [SMulCommClass S R N] [SMulCommClass R S N] :
```

Comment:
You shouldn't need both `SmulCommClass`es here, as one can be built from the other.

Reply:
The following MWE:
```
import Mathlib.Algebra.Module.Defs

variable {S T : Type*}
variable {R : Type*} {M N : Type*}
variable [Semiring S] [CommSemiring R] [AddCommMonoid N] [Module S N] [Module R N] [SMulCommClass S R N]

instance : SMulCommClass R S N := by infer_instance
```
The `infer_instance` fails with `failed to synthesize instance SMulCommClass R S N`.

Reply:
Indeed, but you can build it manually; see 3e887927a25ac161c3ecb10b8f306e3324a4e872

Reply:
Looks like you've fixed this for me now? Thanks.